File: bench/bench.py
import json
from time import time

import flatbush

import pyflatbush

print("loading data")
with open("bounds.txt") as f:
    lines = [json.loads(line) for line in f]


start = time()
index_cy = pyflatbush.Flatbush(len(lines))
for box in lines:
    index_cy.add(*box)

index_cy.finish()
end = time()

print(f"Cython create index: {end - start:.2f}s")

start = time()
index_py = flatbush.FlatBush()
for box in lines:
    index_py.add(*box)

index_py.finish()
end = time()
print(f"Pure Python create index: {end - start:.2f}s")

File: bench/bench.mjs
import fs from 'fs';
import Flatbush from 'flatbush';

var text = fs.readFileSync('bounds.txt', 'UTF-8');
var textLines = text.trim().split('\n');
var bounds = textLines.map(JSON.parse);

console.time('Create index');
// initialize Flatbush
const index = new Flatbush(bounds.length);

// fill it with bounds
for (const p of bounds) {
    index.add(p[0], p[1], p[2], p[3]);
}

// perform the indexing
index.finish();
console.timeEnd('Create index');



// // make a bounding box query
// const found = index.search(minX, minY, maxX, maxY).map((i) => items[i]);

// // make a k-nearest-neighbors query
// const neighborIds = index.neighbors(x, y, 5);

// // instantly transfer the index from a worker to the main thread
// postMessage(index.data, [index.data]);

// // reconstruct the index from a raw array buffer
// const index = Flatbush.from(e.data);
